pkg/forward/forward.go

```diff
@@ -467,17 +571,19 @@ func (isdf *InterStepDataForward) ackFromBuffer(ctx context.Context, offsets []i
 
 // writeToBuffers is a blocking call until all the messages have be forwarded to all the toBuffers, or a shutdown
 // has been initiated while we are stuck looping on an InternalError.
-func (isdf *InterStepDataForward) writeToBuffers(ctx context.Context, messageToStep map[string][]isb.Message) (writeOffsetsEdge map[string][]isb.Offset, err error) {
+func (isdf *InterStepDataForward) writeToBuffers(
```

Comment:
Had to update `writeToBuffers` signature to keep track of previous `writeOffsets`

Reply:
I think it might be better to keep the original signature, because it clearly means the returned offsets are the results of writing the input of messageToStep.

In this case, we merge the returned in offsets in line 488 to the one in line 452 as the function return.